docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -953,12 +953,19 @@ Cython code.  Here is the list of currently supported directives:
     Copy the original source code line by line into C code comments in the generated
     code file to help with understanding the output.
     This is also required for coverage analysis.
-    
+
 ``cpp_locals`` (True / False)
     Make C++ variables behave more like Python variables by allowing them to be
     "unbound" instead of always default-constructing them at the start of a
     function.  See :ref:`cpp_locals directive` for more detail.
 
+``legacy_implicit_noexcept`` (True / False)
+    When enabled, ``cdef`` functions will not propagate raised exceptions by default. Hence,
+    the function will behave same way as if declared with `noexcept` keyword. See
+    :ref:`error_return_values` for details. Setting this directive to ``True`` will
+    cause Cython to have same semantics as Cython 0.2X. This directive is intended to help
+    migrate legacy code written before Cython 3. We recommend leaving it off for new code.
```

Comment:
Since it already raises a deprecation warning, I think this statement could be strengthened:
```suggestion
    cause Cython to have same semantics as Cython 0.2X. This directive is deprecated and
    exists solely to help migrate legacy code written before Cython 3.
```

Reply:
Not sure about it. This text was added after proposal by @da-woods

Reply:
I like "exists solely to help..." - that seems like a mild improvement.

It feels a bit odd to deprecate something before you even add it so I'd possibly rather not use the word "deprecated".

Reply:
Although we use the word "deprecated" elsewhere so maybe its fine.

Reply:
I think this can be marked as resolved after @scoder suggestion